File: Languages/passwordProtect.java
import java.io.*;
public class main {
  public static void main(String[] args){
    part1();
    System.out.println("25% done");
    part2();
    System.out.println("50% done");
    part3();
    System.out.println("75% done");
    part4();
    System.out.println("100% done");
  }
  
  public static void part1(){
    int a=-1;
    while ( a < 9 ) {
      a = a + 1;
      String string_A = String.format("%s", a);
      new File(string_A).mkdir();
    }
  }
  
  public static void part2(){
    int a=-1;
    while ( a < 9 ) {
      a = a + 1;
      int b=-1;
      while ( b < 9 ) {
        b = b + 1;
        String string_AB = String.format("%s/%s", a, b);
        new File(string_AB).mkdir();
      }
    }
  }
  
  public static void part3(){
    int a=-1;
    while ( a < 9 ) {
      a = a + 1;
      int b=-1;
      while ( b < 9 ) {
        b = b + 1;
        int c=-1;
        while ( c < 9 ){
          c = c + 1;
          String string_ABC = String.format("%s/%s/%s", a, b, c);
          new File(string_ABC).mkdir();
        }
      }
    }
  }
  
  public static void part4(){
    int a=-1;
    while ( a < 9 ) {
      a = a + 1;
      int b=-1;
      while ( b < 9 ) {
        b = b + 1;
        int c=-1;
        while ( c < 9 ){
          c = c + 1;
          int d=-1;
          while ( d < 9 ){
            d = d + 1;
            String string_ABCD = String.format("%s/%s/%s/%s", a, b, c, d);
            String file = String.format("%s/%s/%s/%s/.combo", a, b, c, d);
            new File(string_ABCD).mkdir();
            new File(file).createNewFile();
          }
        }
      }
    }
  }
}

File: Languages/passwordProtect.lua
function part1()
  a=-1
  while a < 9
  do
    a=a+1
    os.execute(string.format("mkdir %d", a))
  end
end

function part2()
  a=-1
  while a < 9
  do
    a=a+1
    b=-1
    while b < 9
    do
      b=b+1
      os.execute(string.format("mkdir %d/%d", a, b))
    end
  end
end

function part3()
  a=-1
  while a < 9
  do
    a=a+1
    b=-1
    while b < 9
    do
      b=b+1
      c=-1
      while c < 9
      do
        c=c+1
        os.execute(string.format("mkdir %d/%d/%d", a, b, c))
      end
    end
  end
end

function part4()
  a=-1
  while a < 9
  do
    a=a+1
    b=-1
    while b < 9
    do
      b=b+1
      c=-1
      while c < 9
      do
        c=c+1
        d=-1
        while d < 9
        do
          d=d+1
          os.execute(string.format("mkdir %d/%d/%d/%d", a, b, c, d))
          os.execute(string.format("touch %d/%d/%d/%d/.combo", a, b, c, d))
        end
      end
    end
  end
end

function start()
part1()
print("25% done")
part2()
print("50% done")
part3()
print("75% done")
part4()
print("100% done")
end

start()